rosidl_generator_c/src/string_functions.c

```diff
@@ -25,10 +25,8 @@ rosidl_generator_c__String__init(rosidl_generator_c__String * str)
   if (!str) {
     return false;
   }
-  str->data = malloc(1);
-  if (!str->data) {
-    return false;
-  }
+  assert(str->data = malloc(1));
```

Comment:
I figured that if malloc failed you should probably exit immediately. I didn't know that it would be removed if you compiled in release mode, however. I'll fix that.